homeassistant/components/climate/nuheat.py

```diff
@@ -0,0 +1,212 @@
+        if self._thermostat.heating:
+            return STATE_HEAT
+
+        return STATE_IDLE
+
+    @property
+    def min_temp(self):
+        """Return the minimum supported temperature for the thermostat."""
+        if self._temperature_unit == "C":
+            return self._thermostat.min_celsius
+
+        return self._thermostat.min_fahrenheit
+
+    @property
+    def max_temp(self):
+        """Return the maximum supported temperature for the thermostat."""
+        if self._temperature_unit == "C":
+            return self._thermostat.max_celsius
+
+        return self._thermostat.max_fahrenheit
+
+    @property
+    def target_temperature(self):
+        """Return the currently programmed temperature."""
+        if self._temperature_unit == "C":
+            return self._thermostat.target_celsius
+
+        return self._thermostat.target_fahrenheit
+
+    @property
+    def target_temperature_low(self):
+        """Return the lower bound temperature we try to reach."""
+        return self.target_temperature
+
+    @property
+    def target_temperature_high(self):
+        """Return the upper bound temperature we try to reach."""
+        return self.target_temperature
+
+    @property
+    def current_hold_mode(self):
+        """Return current hold mode."""
+        if self.is_away_mode_on:
+            return MODE_AWAY
+
+        schedule_mode = self._thermostat.schedule_mode
+        if schedule_mode == SCHEDULE_RUN:
+            return MODE_AUTO
+
+        if schedule_mode == SCHEDULE_HOLD:
+            return MODE_HOLD_TEMPERATURE
+
+        if schedule_mode == SCHEDULE_TEMPORARY_HOLD:
+            return MODE_TEMPORARY_HOLD
+
+        return MODE_AUTO
+
+    @property
+    def operation_list(self):
+        """Return list of possible operation modes."""
+        return OPERATION_LIST
+
+    @property
+    def is_away_mode_on(self):
+        """
+        Return true if away mode is on.
+
+        Away mode is determined by setting and HOLDing the target temperature
+        to the minimum temperature supported.
+        """
+        if self._thermostat.target_celsius > self._thermostat.min_celsius:
+            return False
+
+        if self._thermostat.schedule_mode != SCHEDULE_HOLD:
+            return False
+
+        return True
+
+    def turn_away_mode_on(self):
+        """Turn away mode on."""
+        if self.is_away_mode_on:
+            return
+
+        kwargs = {}
+        kwargs[ATTR_TEMPERATURE] = self.min_temp
+
+        self.set_temperature(**kwargs)
+        self._force_update = True
+
+    def turn_away_mode_off(self):
+        """Turn away mode off."""
+        if not self.is_away_mode_on:
+            return
+
+        self.resume_program()
+        self._force_update = True
+
+    def resume_program(self):
+        """Resume the thermostat's programmed schedule."""
+        self._thermostat.resume_schedule()
+        self._force_update = True
+
+    def set_temperature(self, **kwargs):
```

Comment:
According to https://home-assistant.io/components/climate/ , the `set_temperature` service can also be called with a new operation mode, to set operation mode and temperature in one go. This should be handled here, I think.

Reply:
Hmm, I see that, but the only options for this floor heating thermostat are `heat` or `idle`. If you set a `target_temperature` higher than the current `temperature`, it will heat. If you set a `target_temperature` lower than the current `temperature`, it will idle.

In this case, I think that passing in an operation mode would be redundant, a bit confusing, and error prone.

Reply:
But this method could also be used to switch for example from `STATE_AUTO` to `STATE_TEMPERATURE` (and setting a new target temperature at the same time). If you just ignore the operation mode in the kwargs, this will break anything that tries setting the operation mode through this call (which might be the front end, not sure).

Reply:
`STATE_AUTO` and `STATE_TEMPERATURE` are not `operation_modes`, they are `hold_modes`. This method optionally accepts `operation_mode`.

But even so, passing any extra kwargs to this method will not break things, because the `**kwargs` attribute will capture them and ignore them.

The docs you link state that `operation_mode`, `target_temp_high`, and `target_temp_low` are optional and they simply are not relevant for this particular component. The only required attribute is `temperature`, which is in use.

Reply:
Personally, I would prefer if you didn't use the `STATE_*` constants for the hold mode (but only for the operation mode). This will hopefully all soon change, but as far as I can tell, the `STATE_*` constants are meant for operation modes. You can define your own `HOLD_*` constants for your hold modes. I think using `STATE_*` in hold modes makes people (or just me :laughing: ) confuse hold and operation modes even more…

But, back to topic: Do I see this correctly that then the thermostat does have no operation mode? Just hold modes? To me, the exact distinction is not clear, and I would probably have made all of those modes operation modes, but that's no exact science I guess.

Reply:
The thermostat can operate in 2 modes: `heat` and `idle`. On the thermostat itself, this is used simply to report the current operating state, which is what I mirrored for this API.

As far as I understand, `ClimateDevice.current_operation` is supposed to return the currently running operation. ie. heat, cool, idle, etc... which is what I'm doing here.

On the flipside, with NuHeat, the operating mode isn't explicitly controllable except by changing the target temperature relative to the current temperature. That said, `NuHeatThermostat.set_operation_mode` is not implemented, because it's not supported.